TESTING.rst

```diff
@@ -0,0 +1,712 @@
+
+Running Static Code Checks from the Host
+----------------------------------------
+
+You can trigger the static checks from the host environment, without entering the Docker container. To do
+this, run the following scripts (the same is done in Travis CI):
+
+* `<scripts/ci/ci_check_license.sh>`_ - checks the licenses.
+* `<scripts/ci/ci_docs.sh>`_ - checks that documentation can be built without warnings.
+* `<scripts/ci/ci_flake8.sh>`_ - runs Flake8 source code style enforcement tool.
+* `<scripts/ci/ci_lint_dockerfile.sh>`_ - runs lint checker for the Dockerfile.
+* `<scripts/ci/ci_mypy.sh>`_ - runs a check for mypy type annotation consistency.
+* `<scripts/ci/ci_pylint_main.sh>`_ - runs pylint static code checker for main files.
+* `<scripts/ci/ci_pylint_tests.sh>`_ - runs pylint static code checker for tests.
+
+The scripts may ask you to rebuild the images, if needed.
+
+You can force rebuilding the images by deleting the ``.build`` directory. This directory keeps cached
+information about the images already built and you can safely delete it if you want to start from scratch.
+
+After documentation is built, the HTML results are available in the ``docs/_build/html``
+folder. This folder is mounted from the host so you can access those files on your host as well.
+
+Running Static Code Checks in the Docker
+------------------------------------------
+
+If you are already in the Breeze Docker environment (by running the ``./breeze`` command),
+you can also run the same static checks from the container:
+
+* Mypy: ``./scripts/ci/in_container/run_mypy.sh airflow tests``
+* Pylint for main files: ``./scripts/ci/in_container/run_pylint_main.sh``
+* Pylint for test files: ``./scripts/ci/in_container/run_pylint_tests.sh``
+* Flake8: ``./scripts/ci/in_container/run_flake8.sh``
+* License check: ``./scripts/ci/in_container/run_check_licence.sh``
+* Documentation: ``./scripts/ci/in_container/run_docs_build.sh``
+
+Running Static Code Analysis for Selected Files
+-----------------------------------------------
+
+In all static check scripts, both in the container and host versions, you can also pass a module/file path as
+parameters of the scripts to only check selected modules or files. For example:
+
+In the Docker container:
+
+.. code-block::
+
+  ./scripts/ci/in_container/run_pylint.sh ./airflow/example_dags/
+
+or
+
+.. code-block::
+
+  ./scripts/ci/in_container/run_pylint.sh ./airflow/example_dags/test_utils.py
+
+On the host:
+
+.. code-block::
+
+  ./scripts/ci/ci_pylint.sh ./airflow/example_dags/
+
+
+.. code-block::
+
+  ./scripts/ci/ci_pylint.sh ./airflow/example_dags/test_utils.py
+
+Running Test Suites via Scripts
+---------------------------------
+
+To run all tests with default settings (Python 3.6, Sqlite backend, "docker" environment), enter:
+
+.. code-block::
+
+  ./scripts/ci/local_ci_run_airflow_testing.sh
+
+
+To select Python 3.6 version, Postgres backend, and a ``docker`` environment, specify:
```

Comment:
does it mean "edit the .sh script"? or is this an input for the script we launched?